### PR TITLE
Support independent clouds URIs

### PR DESCRIPTION
**Pull request by ArielYehezkely**
#### Pull Request Description

Independent clouds has different endpoint suffix from public cloud.
Replace the hardcoded "core.windows.net" suffix with the actual URL suffix. 

#### Future Release Comment

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- Support independent cloud endpoints URLs suffix.

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/client.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def with_az_cli_authentication(cls, connection_string: str) -> "KustoConnectionS
     def with_aad_managed_service_identity_authentication(
         cls, connection_string: str, client_id: str = None, object_id: str = None, msi_res_id: str = None, timeout: int = None
     ) -> "KustoConnectionStringBuilder":
-        """"
+        """ "
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client.py:329]: there's an extra " here, no?

File: azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py [hunk 2]
```diff
@@ -7,32 +7,33 @@
 from azure.kusto.data import KustoClient
 from azure.kusto.data._models import KustoResultTable
 
-_URI_FORMAT = re.compile("https://(\\w+).(queue|blob|table).core.windows.net/([\\w,-]+)\\?(.*)")
+_URI_FORMAT = re.compile("https://(\\w+).(queue|blob|table).(core.\\w+.net)/([\\w,-]+)\\?(.*)")
```
Comment (yogilad) [inline azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py:10]: for all uses of \w, are we sure dashes and perhaps other characters are not allowed?

File: azure-kusto-data/azure/kusto/data/client.py (hunk 1 above)
Comment (ArielYehezkely) [reply azure-kusto-data/azure/kusto/data/client.py:329]: yep 👍

File: azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py (hunk 2 above)
Comment (ArielYehezkely) [reply azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py:10]: AFAIK other characters are not allowed, but I don't know all independent clouds suffixes..

**Review by yogilad — APPROVED** [review]

**Pull request merged**